fuels-core/src/code_gen/functions_gen.rs

```diff
@@ -209,17 +207,549 @@ fn expand_input_param(
+            inputs: vec![
+                Property {
+                    name: "gas_".to_string(),
+                    type_field: String::from("bool"),
+                    components: None,
+                },
+                Property {
+                    name: "amount_".to_string(),
+                    type_field: String::from("u64"),
+                    components: None,
+                },
+                Property {
+                    name: "color_".to_string(),
+                    type_field: String::from("u32"),
+                    components: None,
+                },
+            ],
+            name: "".to_string(),
+            outputs: vec![],
+        };
+        let hm: HashMap<String, Property> = HashMap::new();
+        let result = expand_function_arguments(&function, &hm, &hm);
+        let (args, call_args) = result.unwrap();
+        let result = format!("({},{})", args.to_string(), call_args.to_string());
+        assert_eq!(result, "(,())");
+    }
+    #[test]
+    fn test_expand_function_arguments_primitive() {
+        let function = Function {
+            type_field: "ZigZag".to_string(),
+            inputs: vec![
+                Property {
+                    name: "BimBam".to_string(),
+                    type_field: String::from("bool"),
+                    components: None,
+                },
+                Property {
+                    name: "".to_string(),
+                    type_field: String::from("u64"),
+                    components: None,
+                },
+            ],
+            name: "PipPop".to_string(),
+            outputs: vec![],
+        };
+        let hm: HashMap<String, Property> = HashMap::new();
+        let result = expand_function_arguments(&function, &hm, &hm);
+        let (args, call_args) = result.unwrap();
+        let result = format!("({},{})", args.to_string(), call_args.to_string());
+        assert_eq!(
+            result,
+            "(, bim_bam : bool , p1 : u64,& [bim_bam . into_token () , p1 . into_token () ,])"
+        );
+    }
+    #[test]
+    fn test_expand_function_arguments_composite() {
+        let function = Function {
+            type_field: "ZigZag".to_string(),
+            inputs: vec![
+                Property {
+                    name: "BimBam".to_string(),
+                    type_field: String::from("struct nameunused"),
+                    // Not parsed, so can be empty but not None
+                    components: Some(vec![]),
+                },
+                Property {
+                    name: "PimPoum".to_string(),
+                    type_field: String::from("enum nameunused"),
+                    // Not parsed, so can be empty but not None
+                    components: Some(vec![]),
+                },
+            ],
+            name: "PipPopFunction".to_string(),
+            outputs: vec![],
+        };
+        let mut custom_structs = HashMap::new();
+        // TODO: inconsistent behavior
```

Comment:
Same for these `TODO: inconsistent behavior`. Let's link them to the external Github issue so we give more context to other people, just in case.

Reply:
same: it will go down with the refactoring, removing the comment.